### ISSUE TITLE
[a11y][Mobile][iOS/android][WCAG 2.4.8 ] [v2.1 AA] Required fields are failing to properly announce it

### ISSUE THREAD

**Issue by dkatzz** [civiform/civiform#9910]
OBSERVED RESULTS:
--

There are some required fields (Marked with a red asterisk) that are announcing it incorrectly while using the screen reader in both android and iOS, verbalizing like "First Name, star" or directly not mentioning the status of required

[iOS](https://drive.google.com/file/d/1BZo2XhScSuLcsZ7NeasU2AV9lC6sSyjw/view?usp=drive_link)

[Android](https://drive.google.com/file/d/1Ej5UeTbWThbGWCrLuc7cOSuLgVgiziRv/view?usp=drive_link)

EXPECTED RESULTS:
--

Required fields are important so providing the correct context to users that rely on screen readers is a priority, something like "FIELDNAME, text field, REQUIRED" is an ideal verbalization to let the user know that he must fill those fields

STEPS TO REPRODUCE:
--
0. Turn on the screen reader (VoiceOver or Talkback )
1. Go to [URL](https://ns.civiform.dev/programs/76/blocks/1/edit)
2. Navigate in a linear manner and notice the verbalization of the required fields that are marked with a red asterisk

RELEVANT GAR SHEET:
--

[GAR sheet](https://docs.google.com/spreadsheets/d/1mgAhVZ3tuJ0O2EeB5JZ0HAvSrxVeIUjlJxVAhu3JH7M/edit?gid=1240238308#gid=1240238308)

ADDITIONAL INFORMATION:
--

[Success Criterion 1.3.5 Identify Input Purpose](https://www.w3.org/TR/WCAG21/#identify-input-purpose)

ENVIRONMENT DETAILS:
--

---
- Devices: iPhone 14 plus: iOS 18.2.1
- Screen reader name: VoiceOver
- Browser name: Chrome Mobile: v133.0.69
---
- Devices: Pixel 7: android 15
- Screen reader name: TalkBack
- Browser name: Chrome Mobile: v133.0.69
---
